### ISSUE TITLE
[readability/braces] False positive for ; after {}

### ISSUE THREAD

**Issue by alejandro-colomar** [cpplint/cpplint#201]
```c
    struct mount_attr *attr = &(struct mount_attr){};
```

I'm not sure if this is valid C++.  It is valid C, for sure.

```
tmp/src/man2/mount_setattr.2.d/mount_setattr.c:61:  You don't need a ; after a }  [readability/braces] [4]
```

The report in this case is invalid, since the `;` is necessary.

**Comment by tkruse** [comment]
Thanks for reporting this. Not sure if it will get fixed, I don't have the time to look into it.

**Comment by georgehosono** [comment]
C++20 Concepts seem to cause the same false positive. The following code (adapted from https://en.cppreference.com/w/cpp/language/constraints) causes the same false positive. The code will not compile without the `;`.

```c++
template <typename T>                                                                              
concept Hashable = requires(T a) {                                                                 
    { std::hash<T>{}(a) } -> std::convertible_to<std::size_t>;                                     
};
```

```
test.h:8:  You don't need a ; after a }  [readability/braces] [4]
```

**Comment by aaronliu0130** [comment]
Note that the concepts issue mentioned by George has been tracked in #225 and fixed last year.

**Comment by YAHYASUFYANI** [comment]
https://github.com/cpplint/cpplint/issues/201#issue-1223407745[](url)
